test/run-tests.py

```diff
@@ -597,15 +641,30 @@ def RunTest(info, options, variables, verbose_level=0):
   os.makedirs(out_dir)
   variables['out_dir'] = out_dir
 
-  cmd = info.GetCommand(rel_gen_input_path, variables, options.arg,
-                        verbose_level)
-  if options.print_cmd:
-    print(' '.join(cmd))
+  total_stdout = ''
+  total_stderr = ''
+  returncode = 0
+  total_duration = 0
 
-  try:
-    return RunCommandWithTimeout(cmd, cwd, timeout, verbose_level > 0, env)
-  except (Error, KeyboardInterrupt) as e:
-    return e
+  for cmd in info.cmds:
+    cmd_list = cmd.AsList(variables, options.arg, verbose_level)
+    if options.print_cmd:
+      print(' '.join(cmd_list))
+
+    try:
+      stdout, stderr, returncode, duration = (
+          RunCommandListWithTimeout(cmd_list, cwd, timeout,
+                                    verbose_level > 0, env))
+      total_stdout += stdout
+      total_stderr += stderr
+      total_duration += duration
+
+      if returncode != 0:
+        break
+    except (Error, KeyboardInterrupt) as e:
```

Comment:
Move the accept up so it only surrounds the RunCommandListWithTimeout call?

Reply:
Done.